src/app/(dashboard)/invoices/page.tsx

```diff
@@ -0,0 +1,421 @@
+				function extractItemsInfo(receiptJson: any) {
+					const items = receiptJson.entities?.productLineItems || [];
+					const extractedInfo = items.map((item: any) => {
+						const itemData = item.data;
+						const itemName = itemData.name?.text || 'Unknown item';
+						const quantity = itemData.quantity?.data || 0;
+						const unitPrice = itemData.unitPrice?.data || 0.0;
+						const totalPrice = itemData.totalPrice?.data || 0.0;
+
+						return {
+							Item: itemName,
+							Quantity: quantity,
+							UnitPrice: `$${unitPrice.toFixed(2)}`,
+							TotalPrice: `$${totalPrice.toFixed(2)}`,
+						};
+					});
+
+					return extractedInfo;
+				}
+
+				// Extract items info
+				const itemsInfo = extractItemsInfo(response);
+				const tmps: any = [];
+				// Assuming you want to print out the extracted information
+
+				itemsInfo.forEach((item: any) => {
+					try {
+						console.log(
+							`Item: ${item.Item}, Quantity: ${item.Quantity}, Unit Price: ${item.UnitPrice}, Total Price: ${item.TotalPrice}`,
+						);
+
+						tmps.push({
+							item: item.Item,
+							Quantity: item.Quantity,
+							price: item.UnitPrice,
+							total_price: item.TotalPrice,
+						});
+						setItems(tmps);
+					} catch (e: any) {
+						alert('failed due to\n:' + e);
+					}
+				});
+
+				setReceiptData(response);
+				setLoading(false);
+				alert(JSON.stringify(response));
+			})
+			.catch(err => {
+				console.error(err);
+				setError('An error occurred while processing the receipt.');
+				setLoading(false);
+			});
+	};
+
+	const handleSubmitForm = async (e: any) => {
+		e.preventDefault();
+
+		// upload image
+		await createReciept(e);
+	};
+	const createReciept = async (e: any) => {
+		// create org using user auth
+		const supabase = await createSupbaseClient();
+
+		// user info
+		const {
+			data: { user },
+			error,
+		} = await supabase.auth.getUser();
+
+		let newURL = null;
+
+		if (imageURL !== DEFAULT_IMAGE) {
+			// create custom hash for image
+			const hash = Math.random().toString(36).substring(2);
+			// upload image to storage
+			const { data, error } = await supabase.storage
+				.from('profile-avatars')
+				.upload(`public/${hash}`, image, {
+					cacheControl: '3600',
+				});
+
+			if (error) {
+				console.error(error);
+				alert(error);
+				return;
+			}
+
+			// get image url
+			const {
+				data: { publicUrl },
+			} = supabase.storage
+				.from('profile-avatars')
+				.getPublicUrl(data?.path as string);
+
+			newURL = publicUrl;
+
+			setImageURL(publicUrl);
+		}
+		reciept.created_by = user?.id;
+		// query to create new row entry
+		const { data: entryData, error: entryError } = await supabase
+			.from('receipts')
+			.insert([
+				{
+					...reciept,
+					image: newURL || imageURL,
+				},
+			]);
+
+		entryError ? alert(JSON.stringify(entryError)) : console.log('');
+		entryError
+			? console.error('error creating receipt ' + entryError)
+			: console.info('Creating reciept worked');
+
+		if (
```

Comment:
span text popup likely if their is any error at all. generalized error messaging indicating to retry or refresh.

Next sprint on this though.